File: Phidget Code/src/Models/In/Match.java
package Models.In;

import java.util.Date;

/**
 * Created by bri_e on 07-04-17.
 */
public class Match {

    private Date startDate;
    private Date endDate;

    public Match(Date begin, Date end){
        startDate = begin;
        endDate = end;
    }

    public boolean areWeDuringMatch(){
        Date today = new Date();

        if (today.after(startDate) && today.before(endDate)) {
            return true;
        } else {
            return false;
        }
    }

    public boolean isMatchFinished(){
        Date today = new Date();
        if (today.after(endDate)){
            return true;
        } else {
            return false;
        }
    }

    @Override
    public String toString(){
        return "Date début : " + startDate + " et date de fin : " + endDate;
    }
}

File: PhidgetScala/src/Models/In/Match.scala
package Models.In

import java.util.Date

/**
  * Created by bri_e on 20-04-17.
  * Class modelling the matches that takes places in the planning
  */
class Match(startDate: Date, endDate: Date) {

  /**
    * @return if we are or not during the match
    */
  def areWeDuringMatch: Boolean = {
    val today = new Date
    if (today.after(startDate) && today.before(endDate)) true
    else false
  }

  /**
    * @return if the match is finished or not
    */
  def isMatchFinished: Boolean = {
    val today = new Date
    if (today.after(endDate)) true
    else false
  }

  override def toString: String = "Date début : " + startDate + " et date de fin : " + endDate
}
